src/components/config.cpp

```diff
@@ -208,6 +211,59 @@ unsigned long long config::convert(string&& value) const {
   return v < ULLONG_MAX ? v : 0ULL;
 }
 
+template <>
+spacing_val config::convert(string&& value) const {
+  char* new_end;
+  auto size_value = std::strtof(value.c_str(), &new_end);
+
+  if (size_value < 0) {
+    throw application_error(sstream() << "Value: " << value << " must be positive ");
+  }
+
+  spacing_val size{spacing_type::SPACE, size_value};
+
+  string unit = string_util::trim(new_end);
+  if (!unit.empty()) {
+    if (unit == "px") {
+      size.type = spacing_type::PIXEL;
+      size.value = std::trunc(size.value);
+    } else if (unit == "pt") {
+      size.type = spacing_type::POINT;
+    } else {
+      size.value = std::trunc(size.value);
+    }
+  }
+
+  return size;
+}
+
+template <>
+extent_val config::convert(std::string&& value) const {
+  return unit_utils::parse_extent(move(value));
+}
+
+/**
+ * Allows a new format for pixel sizes (like width in the bar section)
+ *
+ * The new format is X%:Z, where X is in [0, 100], and Z is any real value
+ * describing a pixel offset. The actual value is calculated by X% * max + Z
+ */
+template <>
+percentage_with_offset config::convert(string&& value) const {
+  size_t i = value.find(':');
+
+  if (i == std::string::npos) {
+    if (value.find('%') != std::string::npos) {
+      return {strtod(value.c_str(), nullptr), {}};
```

Comment:
```suggestion
      return {std::stoi(value), {}};
```

Reply:
This seems to produce a warning
```
[ 41%] Building CXX object bin/CMakeFiles/poly.dir/modules/ipc.cpp.o
/home/frebib/nerdcloud/projects/frebib/polybar/src/components/config.cpp: In member function ‘T polybar::config::convert(std::string&&) const [with T = polybar::percentage_with_offset; std::string = std::__cxx11::basic_string<char>]’:
/home/frebib/nerdcloud/projects/frebib/polybar/src/components/config.cpp:257:24: warning: narrowing conversion of ‘std::__cxx11::stoi((*(const string*)(& value)), 0, 10)’ from ‘int’ to ‘double’ [-Wnarrowing]
  257 |       return {std::stoi(value), {}};
      |               ~~~~~~~~~^~~~~~~
/home/frebib/nerdcloud/projects/frebib/polybar/src/components/config.cpp:263:22: warning: narrowing conversion of ‘std::__cxx11::stoi(percentage, 0, 10)’ from ‘int’ to ‘double’ [-Wnarrowing]
  263 |     return {std::stoi(percentage), convert<extent_val>(value.substr(i + 1))};
      |             ~~~~~~~~~^~~~~~~~~~~~
```

Reply:
I think that was just a brainfart on my side. I wanted to use the c++ stdlib functions for number parsing but probably didn't see that the original produced a double. It should be `std::stod`, I'll change that.